snuba/datasets/transactions_processor.py

```diff
@@ -79,6 +87,9 @@ def process_message(self, message, metadata=None) -> Optional[ProcessedMessage]:
             data["timestamp"],
         )
 
+        duration_secs = (processed["finish_ts"] - processed["start_ts"]).total_seconds()
+        processed['duration'] = int(duration_secs * 1000) if duration_secs > 0 else 0
```

Comment:
You could also express this as `max(int(duration_secs * 1000), 0)` if you think that's clearer (I don't have a strong preference.)